src/hardware/dac.rs

```diff
@@ -39,6 +41,31 @@ pub enum Error {
     Bounds,
 }
 
+/// A type representing a DAC sample.
+#[derive(Copy, Clone, Debug, Format)]
+pub struct DacCode(pub u32);
```

Comment:
The inner `u32` should not be `pub` (also for `AdcCode`). This is fragile. Instead `impl From<DacCode> for u32`.